### DISCUSSION TITLE
Example with multiple parameters?

### DISCUSSION THREAD

**Discussion by theogravity** [daltonmenezes/interprocess#4] [Q&A] [answered]
If I want to define multiple parameters, what's the appropriate way to do it? It seems the `data` field is what you use in a handler for data access, so I was wondering if when you define your interprocess method, would the second param have to be an object if you want to use multiple parameters?

**Comment by daltonmenezes** [comment, marked as answer]
Hi @theogravity , you can pass an object like this:
```ts
import { createIpcSlice } from 'interprocess'

interface Message {
  id: string
  message: string
}

export const rendererIpcSlice = createIpcSlice({
  renderer: {
    async myAnotherIpcHandler(_, { id, message }: Message) {
      console.log({ id, message })

      // do something with the data when this handler is invoked by the main process
      return { message: message.toLowerCase()  }
    },
  },
})
```
